content/en/blog/_posts/Hardware-Accelerated-SSLTLS-Termination-in-Ingress-Controllers-using-Kubernetes-Device-Plugins-and-RuntimeClass.md

```diff
@@ -0,0 +1,168 @@
+resources can be further increased if the PCIe device supports the Single-Root I/O Virtualization
+(SR-IOV) specification. SR-IOV allows the PCIe device to be split further to *virtual functions* (VF),
+derived from *physical function* (PF) devices, and each belonging to their own IOMMU group. To expose
+these IOMMU isolated device functions to user space and containers, the host kernel should bind
+them to a specific device driver. In Linux, this  driver is vfio-pci and it makes each device
+available through a character device in user space. The kernel vfio-pci driver provides user space
+applications with a direct, IOMMU backed access to PCIe devices and functions, using a mechanism
+called *PCI passthrough*. The interface can be leveraged by user space frameworks, such as the
+Data Plane Development Kit (DPDK). Additionally, virtual machine (VM) hypervisors can provide
+these user space device nodes to VMs and expose them as PCI devices to the guest kernel.
+Assuming support from the guest kernel, the VM gets close to native performant direct access to the
+underlying host devices.
+
+To advertise these device resources to Kubernetes, we can have a simple Kubernetes device plugin
+that runs the initialization (i.e., binding), calls kubelet’s `Registration` gRPC service, and
+implements the DevicePlugin gRPC service that kubelet calls to, e.g., to `Allocate` the resources
+upon Pod creation.
+
+## Device Assignment and Pod Deployment
+
+At this point, you may ask what the container could do with a VFIO device node? The answer comes
+after we first take a quick look into the Kubernetes RuntimeClass.
+
+The Kubernetes RuntimeClass was created to provide better control and configurability
+over a variety of *runtimes* (an earlier [blog post](https://kubernetes.io/blog/2018/10/10/kubernetes-v1.12-introducing-runtimeclass/) goes into the details of the needs,
+status and roadmap for it) that are available in the cluster. In essence, the RuntimeClass
+provides cluster users better tools to pick and use the runtime that best suits for the pod use case.
+
+The OCI compatible [Kata Containers runtime](https://katacontainers.io/) provides workloads with a hardware virtualized
+isolation layer. In addition to workload isolation, the Kata Containers VM has the added
+side benefit that the VFIO devices, as `Allocate`’d by the device plugin, can be passed
+through to the container as hardware isolated devices. The only requirement is that the
+Kata Containers kernel has driver for the exposed device enabled.
+
+That’s all it really takes to enable hardware accelerated crypto for container workloads. To summarize:
+
+  1. Cluster needs a device plugin running on the node that provides the hardware,
+  2. Device plugin exposes the hardware to user space using  the VFIO driver,
+  3. Pod requests the device resources and Kata Containers as the RuntimeClass in the PodSpec,
```

Comment:
Would omit the `commas` after each sentence and could use an unordered list.

Reply:
dropped the commas